examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,314 @@
+# squared coefficient.
+
+from sklearn.metrics import median_absolute_error
+
+y_pred = model.predict(X_train)
+mae = median_absolute_error(y_train, y_pred)
+string_score = 'MAE on training set: {0:.2f} $/hour'.format(mae)
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+r2score = model.score(X_test, y_test)
+
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+string_score += '\nR2 score: {0:.4f}'.format(r2score)
+fig, ax = plt.subplots(figsize=(6, 6))
+sns.regplot(y_test, y_pred)
+
+plt.text(3, 20, string_score)
+
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The model learnt is far from being a good model making accurate predictions:
+# the R squared score is very low.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
+# In this case, we are more interested in providing a methodology than in
+# having a good description of the data: a bad example illustrates the
+# importance of cross checking the results.
+#
+# Interpreting coefficients
+# .........................
+#
+# First of all, we can plot the values of the coefficients of the regressor we
+# have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, binary_columns, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since we did
+# not scale the data before the fit, features having different value ranges.
+# For instance, the AGE coefficient is expressed in $/hours/leaving years
+# while the EDUCATION is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std().plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then normalize the coefficients by the standard deviation and we
+# will be able to compare them and helps interpretation: the greater the
+# variance of a feature, the large the impact of the corresponding coefficent
+# on the output.
```

Comment:
large -> larger

But I don't think the sentence is correct anyway. Take a purely random feature with very high variance: it has zero predictive power, so it's coeff will be close to 0, and the effect of this coeff on the output will be negligible.

Reply:
Definitely @GaelVaroquaux could you clarify? This comes from [here](http://gael-varoquaux.info/interpreting_ml_tuto/content/02_why/01_interpreting_linear_models.html#coefficients-of-a-linear-model).
Thanks.

Reply:
I think the original intent was to say that the bigger the scale of a feature, the bigger the coefficient, **all else being equal.**

Reply:
Yes, my phrasing was bad.

What's important is to compare features within their natural scaling. If I have age, from 20 to 80, and sex, which is 0 or 1, they have a very different scale.